crates/holochain/tests/tests/websocket/mod.rs

```diff
@@ -640,6 +640,22 @@ async fn connection_limit_is_respected() {
     // Disconnect all the clients
     handles.clear();
 
+    // Wait up to 2 seconds for new connections to be permitted
+    let now = Instant::now();
+    while now.elapsed() < Duration::from_secs(2) {
+        let (sender, rx) = connect(cfg.clone(), addr).await.unwrap();
+        let _rx = WsPollRecv::new::<AdminResponse>(rx);
+
+        // Getting a sender back isn't enough to know that the connection succeeded because the other side takes a moment to shutdown, try sending to be sure
+        if sender
+            .request::<AdminRequest, AdminResponse>(AdminRequest::ListDnas)
+            .await
+            .is_ok()
+        {
+            break;
+        }
```

Comment:
Do you want to go ahead and put this one in the `handles` variable, and adjust the loop below to be `while handles.len() < MAX_CONNECTION` just so we don't have to wait for this one to be dropped too?

Reply:
That makes a lot of sense, done